src/Compilers/CSharp/Test/Emit/Attributes/AttributeTests_RefReadOnly.cs

```diff
@@ -0,0 +1,642 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using Microsoft.CodeAnalysis.CSharp.Symbols;
+using Microsoft.CodeAnalysis.CSharp.Test.Utilities;
+using Microsoft.CodeAnalysis.CSharp.UnitTests;
+using Roslyn.Test.Utilities;
+using Microsoft.CodeAnalysis.Test.Utilities;
+using System.Linq;
+using Xunit;
+using Microsoft.CodeAnalysis.CSharp.Symbols.Metadata.PE;
+
+namespace Microsoft.CodeAnalysis.CSharp.UnitTests
+{
+    public class AttributeTests_RefReadOnly : CSharpTestBase
+    {
+        [Fact]
+        public void RefReadOnlyIsWrittenToMetadata_SameAssembly_Method()
+        {
+            var text = @"
+namespace System.Runtime.CompilerServices
+{
+    public class ReadOnlyAttribute : System.Attribute { }
+}
+class Test
+{
+    public ref readonly int M(ref readonly int x) { return ref x; }
+}
+";
+
+            CompileAndVerify(text, verify: false, symbolValidator: module =>
+            {
+                var method = module.ContainingAssembly.GetTypeByMetadataName("Test").GetMethod("M");
+                Assert.Equal(RefKind.RefReadOnly, method.RefKind);
+                Assert.True(method.ReturnsByRefReadonly);
```

Comment:
We have not discussed the exact API for the ref readonly member. For now we have followed the same pattern as with regular ref returning members, which are identified via "ReturnsByRef"

---
In reply to: [110281201](https://github.com/dotnet/roslyn/pull/18358#discussion_r110281201) [](ancestors = 110281201)